libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt

```diff
@@ -18,24 +18,19 @@ suspend fun parseDirectionsResponse(
 ): Expected<Throwable, List<DirectionsRoute>> =
     withContext(dispatcher) {
         return@withContext try {
-            val jsonObject = JSONObject(json)
-            val uuid: String? = if (jsonObject.has(UUID)) {
-                jsonObject.getString(UUID)
-            } else {
-                null
-            }
-
-            // TODO remove after https://github.com/mapbox/navigation-sdks/issues/1229
-            if (jsonObject.has(METADATA)) {
-                onMetadata(jsonObject.getString(METADATA))
+            // TODO simplify when https://github.com/mapbox/mapbox-java/issues/1292 is finished
+            val response = DirectionsResponse.fromJson(json)
```

Comment:
afair this method doesn't handle uuid and we need to use `DirectionsResponse.fromJson(json, options, uuid)`
cc @LukasPaczos

Reply:
`uuid` is handled manually below.

Reply:
@zeac yes, you take it as `val uuid = response.uuid()`
but afaik response will always have `null` uuid, if it is parsed with  `DirectionsResponse.fromJson(json)`